### ISSUE TITLE
Please add support for combined libLLVM

### ISSUE THREAD

**Issue by StefanBruens** [imageworks/OpenShadingLanguage#963]
### Problem

libLLVM can either be built as component libraries or as combined libraries, see
https://wiki.gentoo.org/wiki/Project:LLVM/Shared_libraries

When LLVM is built as combined library, there is only libLLVM.so, and building
oslexec fails, as it uses LLVM_MCJIT_LIBRARY:

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
LLVM_MCJIT_LIBRARY
    linked by target "oslexec" in directory /home/stefan/Documents/Sources/OpenShadingLanguage/src/liboslexec
```

**Expected behavior:**  

LLVM_MCJIT_LIBRARY should be set to a useful value. When using llvm-config, the correct library is returned:

```
$> llvm-config --libfiles mcjit
/usr/lib64/libLLVM.so
llvm-config --libs mcjit
-lLLVM
```

The issue can be worked around by setting "-DLLVM_MCJIT_LIBRARY='' " when running cmake.

### Versions

* OSL branch/version: 1.9.3, master
* OS: openSUSE TW 20190109
* C++ compiler: gcc 8.2
* LLVM version: 7.0.0
* OIIO version:

**Comment by lgritz** [comment]
I'm not quite sure how to do this, and it's hard for me to test because all of my LLVM installs are the component library setup, I guess?

Since you seem to be in the situation of needing it, I don't suppose you can submit a PR that contains the necessary changes to our cmake scripts that make it work? (And, obviously, needs to continue working for those with the component library setup as well.)

**Comment by StefanBruens** [comment]
It should be trivial:
- remove both find_library(LLVM ...) and find_library(LLVMMCJIT ...) from FindLLVM.cmake
- run the following when LLVM_LIBRARIES/LLVM_LIBRARY is not set (copied from the existing code):
```
if ((NOT LLVM_LIBRARY) AND (NOT LLVM_LIBRARIES))
    execute_process (COMMAND ${LLVM_CONFIG} --libfiles
                     OUTPUT_VARIABLE LLVM_LIBRARIES
                     OUTPUT_STRIP_TRAILING_WHITESPACE)
    string (REPLACE " " ";" LLVM_LIBRARIES "${LLVM_LIBRARIES}")
    set (LLVM_LIBRARY "")
endif()
```
`llvm-config --libfiles` generates the correct file list for me for both component and combined library. I.e. it generates many entries for split LLVM and a single entry otherwise.

**Comment by laerne** [comment]
Hello.  I have the same issue on Fedora 29 here : everything is bundled in libLLVM.so.  Adding the above code right at the end (before the message(...)) did solve the issue.  However the handy confifuration variables to change the library to use in ccmake/cmake-gui disappear which I guess is can issue on windows where one often install dependencies to a custom location.

**Comment by lgritz** [comment]
I proposed #976 before noticing this old issue which it should fix. I didn't do it quite the way you suggest, but @StefanBruens can you try out #976 and see if it looks ok and/or works for you?

**Comment by laerne** [comment]
Woops, spoke too fast.  The solution presented hear makes me have a bunch of link errors with a test executable and LLVM at compilation.  I do not advise it.

**Comment by lgritz** [comment]
@laerne, can you also check #976, which is a slightly different approach to solving this problem?

**Comment by lgritz** [comment]
Fixed by #976
